sympy/physics/control/tests/test_lti.py

```diff
@@ -177,9 +182,20 @@ def test_TransferFunction_functions():
     # purely symbolic polynomials.
     p1 = a1*s + a0
     p2 = b2*s**2 + b1*s + b0
-    tf_ = TransferFunction(p1, p2, s)
+    SP1 = TransferFunction(p1, p2, s)
     expect = TransferFunction(2*s + 1, 5*s**2 + 4*s + 3, s)
-    assert tf_.evalf(subs={a0: 1, a1: 2, b0: 3, b1: 4, b2: 5}) == expect
+    assert SP1.evalf(subs={a0: 1, a1: 2, b0: 3, b1: 4, b2: 5}) == expect
```

Comment:
@moorepants, the description of `evalf` says:

```
    Adaptive numerical evaluation of SymPy expressions, using mpmath
    for mathematical functions
```
Are we sure that this would be used for symbolic evaluation in our case? 
``` python
>>> p5 = a*s + b
>>> SP3 = TransferFunction(p5, p, s)
>>> assert SP3.evalf(subs={a: 3, b: -4}) == TransferFunction(3*s - 4, p, s)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'TransferFunction' object has no attribute 'evalf'
```
Should I define a new method `evalf` here?

Reply:
If you want to use `evalf`, you should inherit `EvalfMixin` class and define some necessary methods for it.

Reply:
Yes, subs, xreplace, evalf (and any other similar sympy methods) should work.

Reply:
IMO, we should not inherit another `EvalfMixin` class. I think the more superclasses `TransferFunction` inherits from, the more maintenance/changes are needed. 

Isn't `evalf` normally used for numerical evaluation?
I don't see any difference between `evalf` and normal usage of `subs` here:
``` python
>>> a0,a1,b0,b1,b2=symbols('a0:2, b0:3')
>>> p1 = a1*s + a0
>>> p2 = b2*s**2 + b1*s + b0
>>> SP1=TransferFunction(p1,p2,s)
>>> SP1.subs({a0:1, a1: 2, b0:3, b1: 4, b2: 5})
TransferFunction(2*s + 1, 5*s**2 + 4*s + 3, s)
>>> SP1.evalf(subs={a0: 1, a1: 2, b0: 3, b1: 4, b2: 5})
TransferFunction(2*s + 1, 5*s**2 + 4*s + 3, s)
```

Reply:
Mixing in evalf class is fine and common. That's what it is there for. evalf and subs serve different purposes and have distinct functionally. Both need to be supported.